ydb/public/lib/ydb_cli/commands/interactive/yql_highlight.cpp

```diff
@@ -0,0 +1,71 @@
+#include "yql_highlight.h"
+
+#include <contrib/libs/antlr4_cpp_runtime/src/antlr4-runtime.h>
+#include <ydb/public/lib/ydb_cli/commands/interactive/antlr/YQLLexer.h>
+
+namespace NYdb {
+    namespace NConsoleClient {
+
+        YQLHighlight::ColorSchema YQLHighlight::ColorSchema::Default() {
+            return {
+                .keyword = Color::BLUE,
+                .operation = Color::YELLOW,
+                .identifier = Color::RED,
+                .string = Color::GREEN,
+                .number = Color::CYAN,
+                .unknown = Color::DEFAULT,
+            };
+        }
+
+        namespace {
+
+            YQLHighlight::Color ColorOf(const YQLHighlight::ColorSchema& schema,
+                                        const antlr4::Token* token) {
+                const auto type = token->getType();
+                if (type == YQLLexer::STRING_VALUE) {
+                    return schema.string;
+                }
+                if (YQLLexer::ID_PLAIN <= type && type <= YQLLexer::ID_QUOTED) {
```

Comment:
Why <= and not ==? Looks weird

Reply:
I agree. Wanted to be consistent with this range checks. Also, I have an idea to add static assertions here to make sure that we have covered all token types.

Reply:
Not a big deal if we won't cover some new type. It will just be white, right?
But with this code we rely on enum's actual values. Not sure if its good at all. It is generated, right? So if somebody changes the order of type declarations in grammar file (which is not restricted), values will change and this code may be broken?

Reply:
Yes, if someone changes the order of definitions in the grammar, the code will stop working. To avoid this, I can replace this with a large switch case that uses all tokens.

Reply:
> To avoid this, I can replace this with a large switch-case that uses all tokens.

Looks like there is no other choice. We show explicitly what token types we want to color. And do not rely on the order of declaration appearance in the grammar.

A default option should be everything unknown, with a default color. So that we will notice it if we miss a type or a new type will appear.

Reply:
@pnv1, I rewrote `ColorOf` function to use switch-case. 

There is no reason to make it in a single `switch-case`, because we can't check it for an exhaustiveness, as ANTLR4-generated Lexer stores token indexes in an anon enum. So I decided to split it into functions as in future there can be added more logic like highlighting table and column names differently.

It should be optimized into huge RO table lookup (https://godbolt.org/z/x3avjEdaG).

The other option is to use `std::unordered_map` that can be less performant.

Reply:
Yeah, enum is huge now. But since we have no other choice, I think its good